### PR TITLE
fix: update user management filters for emails with special characters

### PR DESCRIPTION
**Pull request by ss-nikunj**
## Description

This PR fixes an issue where the user management filters were not working for email addresses containing special characters like `+` . Examples of problematic emails include:

- `nikunj.maniya+servicemanager@cofinity-x.com`
- `nikunj+CompanyAdmin@gmail.com`

- App Access Management -> Add Role -> User List
App Access Management -> Add Role -> User List have same special character (+) related issue on search input.
## Why
- Updated the email filter logic to correctly handle special characters such as `+`.

## Issue
- Issue: [#1128](https://github.com/eclipse-tractusx/portal-frontend/issues/1128)

## Checklist
- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have created and linked IP issues or requested their creation by a committer
- [x] I have created and linked IP issues or requested their creation by a committer
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have checked that new and existing tests pass locally with my changes

image
image

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -40,8 +40,12 @@
 - **Technical User Management**
   - Display technicalUserManagement button based on role validation [#1073](https://github.com/eclipse-tractusx/portal-frontend/pull/1073)
 - **OSP Consent form**
+
```
Comment (JordanGerada3) [inline CHANGELOG.md:43]: Can we remove this introduced space

**Review by JordanGerada3 — APPROVED** [review]
locally tested, works good!

**Comment by ss-nikunj** [comment]
Hi @manojava-gk, 
As per your suggestions I have moved Search email regex into Patterns file. Can you please review it again. 

Thank you!

**Comment by ss-nikunj** [comment]
> Move regex to patterns file

Hello @manojava-gk, 

Regex moved into Patterns file and I have added testcases for this newly added pattern. Can you please review it again. 

Thank you!!

**Comment by evegufy** [comment]
@manojava-gk could you please review again?

**Review by manojava-gk — APPROVED** [review]
looks good to me

File: src/components/shared/frame/UserList/index.tsx [hunk 2]
```diff
@@ -87,7 +88,7 @@ export const UserList = ({
   const searchInputData = useSelector(appManagementSelector)
 
   const validateSearchText = (expr: string) => {
-    const validateExpr = /^[ A-Za-z0-9._!@-]*$/.test(expr)
+    const validateExpr = Patterns.EMAIL_SEARCH.test(expr)
```
Comment (oyo) [inline src/components/shared/frame/UserList/index.tsx:91]: I guess this should better use the function `isSearchUserEmail(expr)` from Patterns.ts below

**Review by oyo — CHANGES_REQUESTED** [review]
one small change

File: src/components/shared/frame/UserList/index.tsx (hunk 2 above)
Comment (ss-nikunj) [reply src/components/shared/frame/UserList/index.tsx]: Hi @oyo, Can you please review it? 
I made a changes according to your suggestions.

**Comment by ss-nikunj** [comment]
Hi @oyo, I made a changes according to your feedback. Please review it again.

**Review by oyo — APPROVED** [review]

**Pull request merged**